assets/js/imdbae/chat/chatbox.jsx

```diff
@@ -0,0 +1,24 @@
+import React from 'react';
+import socket from '../../socket';
+
+function submitMsg(ev) {
+    console.log("SUBMITTED");
+    ev.preventDefault();
+}
+
+export const Chatbox = (props) => {
```

Comment:
To get the dispatch and the state, you'll have to `connect this component, and map them to props like so:

```javascript
const mapStateToProps => state => {
  // here you have access to the entire state, but you might want to just pass the necessary child states
  return {
    chat: state.chat // this can be accessed from the component with 'props.chat'
  }
};

const mapDispatchToProps => dispatch => {
  // here you have access to the dispatch func, but you might want to map the actions this component will make to make things cleaner and easier
  return {
   // will be mapped to 'props.submitMsg('a message')
    submitMsg: message => dispatch({type: 'SUBMIT_MESSAGE', message}) 
  }
};
```

Finally, you can connect the component:

```javascript
export const Chatbox = connect(mapStateToProps, mapDispatchToProps)((props) => {
  // ... the rest of the Chatbox component
});
```